sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastNestedLoopJoinExec.scala

```diff
@@ -358,6 +339,34 @@ case class BroadcastNestedLoopJoinExec(
     )
   }
 
+  /**
+   * Get matched rows from broadcast side as a [[BitSet]].
+   * Create a local [[BitSet]] for broadcast side on each RDD partition,
+   * and merge all [[BitSet]]s together.
+   */
+  private def getMatchedBroadcastRowsBitSet(
+      streamRdd: RDD[InternalRow],
```

Comment:
we need this variable `streamRdd ` here? How about referencing `streamed` directly?

Reply:
@maropu - then we need to call `streamed.execute()`twice in `defaultJoin()`. I feel it's bit weird and not sure if it can lead to any bug in the future. Normally during execution, we only call `child.execute()` once.